src/devices/QwiicButton/samples/FindI2cDevices.cs

```diff
@@ -0,0 +1,61 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+
+namespace Iot.Device.QwiicButton.Samples
+{
+    /// <summary>
+    /// Finds the connected I2C devices' addresses.
+    /// </summary>
+    internal class FindI2cDevices
+    {
+        public static void Run()
+        {
+            Console.WriteLine("Enter I2C bus ID to scan: [Press Enter for default = 1]");
+            var busIdAsString = Console.ReadLine();
+            if (string.IsNullOrWhiteSpace(busIdAsString))
+            {
+                busIdAsString = "1";
+            }
+
+            if (!Int32.TryParse(busIdAsString, out int i2cBusId))
+            {
+                Console.WriteLine("You must enter an integer - exiting...");
+                return;
+            }
+
+            Console.WriteLine($"Scanning I2C bus {i2cBusId}...");
+            int devicesFoundCount = 0;
+
+            for (int address = 8; address < 120; address++)
+            {
+                try
+                {
+                    using (var button = new QwiicButton(i2cBusId, (byte)address))
+                    {
+                        // Try to read device ID - fails if no device on address
+                        button.GetDeviceId();
+                    }
+
+                    devicesFoundCount++;
+                    Console.WriteLine($"Found I2C device on address 0x{BitConverter.ToString(new[] { (byte)address })} ({address})");
```

Comment:
Just use "{address:X}"

Reply:
Also, be more precise in the text: You're only searching Qwiic buttons, there might be other devices on the bus.

Reply:
Hex notation: fixed.
As for the text: The way I detect buttons on the bus is by trying to read the register at address 0x00 and unfortunately other I2C devices could potentially contain something on that address as well, which means I cannot be completely sure that I found a button... That's why I use the words "I2C device".